File: engine/franca/java/graphics/animations/Ease.java
package franca.java.graphics.animations;

import franca.java.expected.TranspilableClass;
import franca.java.expected.Router;

public class Ease extends TranspilableClass {

  public static final int EASE_LINEAR = 1;

  public int initialValue;
  public int currentValue;
  public int targetValue;

  public long duration;

  public Ease(int initialValue, int targetValue, long duration) {
    super();

    this.initialValue = initialValue;
    this.currentValue = this.initialValue;
    this.targetValue = targetValue;

    this.duration = duration;
  }

  public boolean isRunning() {
    return currentValue != targetValue;
  }

  public boolean tick(Router router, long startedTime) {
    // returns true when needs repainting
    return true;
  }
}

File: platforms/typescript/src/java-franca/animations/ease.ts
import {Router} from "../device/router";

export class Ease {
  static readonly EASE_LINEAR: number = 1;

  initialValue: number;
  currentValue: number;
  targetValue: number;
  duration: number;

  constructor(initialValue: number, targetValue: number, duration: number) {
    this.initialValue = initialValue;
    this.currentValue = this.initialValue;
    this.targetValue = targetValue;
    this.duration = duration;
  }

  destroy(): void {
    // очистка, если нужна
  }

  isRunning(): boolean {
    return this.currentValue !== this.targetValue;
  }

  tick(router: Router, startedTime: number): boolean {
    // returns true when needs repainting
    return true;
  }
}
